lib/util/ast.js

```diff
@@ -201,6 +201,16 @@ function getFirstNodeInLine(context, node) {
   return token;
 }
 
+/**
+ * Get the scope of a context bound node
+ * @param {Context} context
+ * @param {ASTNode} node
+ * @return {Scope}
+ */
+function getScope(context, node) {
+  return (context.sourceCode || context).getScope(node);
```

Comment:
let’s rework this so it checks for context.sourceCode.getScope specifically, so it doesn’t break on versions of eslint with .sourceCode but not .sourceCode.getScope

Reply:
Done.